torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,467 @@
+            for modality_link in self.modality_urls[modal]:
+                if modality_link.endswith("csv"):
+                    self._verify(url=modality_link, md5=None)
+                else:
+                    self._verify(
+                        url=modality_link,
+                        md5=self.md5s[os.path.split(modality_link)[1]],
+                    )
+
+            #  Merge modalities downloaded and extracted in two parts.
+            if modal not in os.listdir(root) and len(self.modality_urls[modal]) == 2:
+                self._merge_parts(root, modal)
+
+        # Masks will be loaded seperately in the :meth:`__getitem__`
+        if "mask" in self.modality:
+            self.modality.remove("mask")
+
+        if "split_ids" in self.modality:
+            self.modality.remove("split_ids")
+
+        if os.path.exists(os.path.join(self.root, "split_IDs.csv")):
+            split_dataframe = pd.read_csv(os.path.join(self.root, "split_IDs.csv"))
+            self.ids = split_dataframe[split].dropna().values.tolist()
+            self.ids = [int(i) for i in self.ids]
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        list_modals = []
+        id = self.ids[index]
+
+        mask = self._load_raster(id, "mask")
+        mask[mask != 0] = 1
+
+        for mode in self.modality:
+            mode = mode.upper() if mode in ["esa_wc", "viirs"] else mode
+            data = self._load_raster(id, mode)
+            list_modals.append(data)
+
+        image = torch.cat(list_modals, dim=0)
+
+        if self.transforms is not None:
+            sample_: dict[str, Tensor] = {"image": image, "mask": mask}
+
+            transformed = self.transforms(sample_)
+
+            image = Rearrange("h w c -> c h w")(transformed["image"])
+            mask = Rearrange("h w c -> c h w")(transformed["mask"])
+
+        sample: dict[str, Tensor] = {"image": image, "mask": mask}
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.ids)
+
+    def _load_raster(self, filename: int, source: str) -> Tensor:
+        """Load a single raster image or target.
+
+        Args:
+            filename: name of the file to load
+            source: the filename of the modality
+
+        Returns:
+            the raster image or target
+        """
+        with rasterio.open(os.path.join(self.root, source, f"{filename}.tif")) as f:
+            array = f.read()
+            if array.dtype == np.uint16:
+                array = array.astype(np.int32)
+            tensor = torch.from_numpy(array)
+            return tensor
+
+    def _verify(self, url: str, md5: Optional[str] = None) -> None:
```

Comment:
Normally these methods don't have any arguments, just a for-loop that checks/downloads/extracts all parts.

Reply:
Would it be fine to keep it this way? I somehow needed to start off the PR this way to have higher degree of freedom in the __init__. Could try to change it now though.

Reply:
It would be fine. If we ever need to force all of them to have the same signature I can update this to match in the future.